x/ecocredit/server/helpers.go

```diff
@@ -56,13 +55,18 @@ func getSubAndSetDecimal(store sdk.KVStore, key []byte, x *apd.Decimal) error {
 		return err
 	}
 
-	err = math.SafeSub(value, value, x)
-	if err != nil {
-		return err
+	if value.IsZero() {
```

Comment:
shouldn't we check for zero value once we have substracted `x` from `value`?

Reply:
sorry, my bad. I'll fix it

Reply:
sorry, my bad. I'll fix it